aiowebostv/webos_client.py

```diff
@@ -114,31 +118,45 @@ def registration_msg(self) -> dict[str, Any]:
 
     async def _ws_connect(
         self, uri: str, ssl_context: ssl.SSLContext | None
-    ) -> WebSocketClientProtocol:
+    ) -> ClientWebSocketResponse:
         """Create websocket connection."""
         _LOGGER.debug("connect(%s): uri: %s", self.host, uri)
-        return await ws_connect(
-            uri,
-            ping_interval=self.ping_interval,
-            ping_timeout=self.ping_timeout,
-            open_timeout=self.timeout_connect,
-            close_timeout=self.timeout_connect,
-            max_size=None,
-            ssl=ssl_context,
-        )
+
+        if TYPE_CHECKING:
+            assert self.client_session is not None
+
+        async with asyncio.timeout(self.timeout_connect):
+            return await self.client_session.ws_connect(
+                uri, heartbeat=self.heartbeat, ssl=ssl_context
+            )
+
+    async def close_client_session(self) -> None:
+        """Close client session if we created it."""
+        if TYPE_CHECKING:
+            assert self.client_session is not None
+
+        await self.client_session.close()
```

Comment:
Only close when the client session is created by the lib

Reply:
https://github.com/home-assistant-libs/aiowebostv/blob/3e739e38cd12356d402f2df8891b9ba5ff16fde6/aiowebostv/webos_client.py#L276-L277

Reply:
Oh lol, I saw the callback, but I didn't see the if statement